ui/app/models/kv/data.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Model, { attr } from '@ember-data/model';
+import lazyCapabilities, { apiPath } from 'vault/macros/lazy-capabilities';
+import { withModelValidations } from 'vault/decorators/model-validations';
+import { withFormFields } from 'vault/decorators/model-form-fields';
+
+/* sample response
+{
+  "data": {
+    "data": {
+      "foo": "bar"
+    },
+    "metadata": {
+      "created_time": "2018-03-22T02:24:06.945319214Z",
+      "custom_metadata": {
+        "owner": "jdoe",
+        "mission_critical": "false"
+      },
+      "deletion_time": "",
+      "destroyed": false,
+      "version": 2
+    }
+  }
+}
+*/
+
+const validations = {
+  path: [
+    { type: 'presence', message: `Path can't be blank.` },
+    {
+      validator: (model) => (model.path?.match(/\/$/) ? false : true),
```

Comment:
as a follow on this could be cleaned up to use the validator we added `endsInSlash`
```suggestion
      type: 'endsInSlash',
```

Reply:
@hellobontempo